### PR TITLE
Display sent message pill on the service dashboard

### PR DESCRIPTION
**Pull request by jimleroyer**
# Summary | Résumé

Admin changes to add the 'sent' notification status.

# Test instructions | Instructions pour tester la modification

_TODO_

# Help requested | Aide requise

I will need designer help to review the changes.

# Unresolved questions / Out of scope | Questions non résolues ou hors sujet

* Do we want to go forward with this change?

# Reviewer checklist | Liste de vérification du réviseur

This is a suggested checklist of questions reviewers might ask during their
review | Voici une suggestion de liste de vérification comprenant des questions
que les réviseurs pourraient poser pendant leur examen :

- [ ] Does this meet a user need? | Est-ce que ça répond à un besoin utilisateur?
- [ ] Is it accessible? | Est-ce que c’est accessible?
- [ ] Is it translated between both offical languages? | Est-ce dans les deux
      langues officielles?
- [ ] Is the code maintainable? | Est-ce que le code peut être maintenu?
- [ ] Have you tested it? | L’avez-vous testé?
- [ ] Are there automated tests? | Y a-t-il des tests automatisés?
- [ ] Does this cause automated test coverage to drop? | Est-ce que ça entraîne
      une baisse de la quantité de code couvert par les tests automatisés?
- [ ] Does this break existing functionality? | Est-ce que ça brise une
      fonctionnalité existante?
- [ ] Should this be split into smaller PRs to decrease change risk? | Est-ce
      que ça devrait être divisé en de plus petites demandes de tirage (« pull
      requests ») afin de réduire le risque lié aux modifications?
- [ ] Does this change the privacy policy? | Est-ce que ça entraîne une
      modification de la politique de confidentialité?
- [ ] Does this introduce any security concerns? | Est-ce que ça introduit des
      préoccupations liées à la sécurité?
- [ ] Does this significantly alter performance? | Est-ce que ça modifie de
      façon importante la performance?
- [ ] What is the risk level of using added dependencies? | Quel est le degré de
      risque d’utiliser des dépendances ajoutées?
- [ ] Should any documentation be updated as a result of this? (i.e. README
      setup, etc.) | Faudra-t-il mettre à jour la documentation à la suite de ce
      changement (fichier README, etc.)?

### CODE REVIEW

**Comment by yaelberger-commits** [comment]
50% done through PR, but requires some UX interaction.

File: app/templates/partials/nav/gc_header_nav.html [hunk 1]
```diff
@@ -77,7 +77,7 @@
               {% endif %}
               {{ nav_menu_item( url_for('main.choose_template', service_id=current_service.id),_('Templates'),'header--'+header_navigation.is_selected('templates')) }}
               {% if not current_user.has_permissions('view_activity') %}
-                {{ nav_menu_item(url_for('main.view_notifications', service_id=current_service.id, status='sending,delivered,failed'),_('Sent messages'),'header--'+header_navigation.is_selected('sent-messages')) }}
+                {{ nav_menu_item(url_for('main.view_notifications', service_id=current_service.id, status='sending,sent,delivered,failed'),_('Sent messages'),'header--'+header_navigation.is_selected('sent-messages')) }}
```
Comment (amazingphilippe) [inline app/templates/partials/nav/gc_header_nav.html:80]: What is this menu item? I've never seen it. What users have the "view_activity" permissions?

**Review by amazingphilippe — COMMENTED** [review]
I think this is a valid option. Curious to release it and see if it clear things up on the support side. My main concern is that it might be unclear what the difference between sent and delivered is. Link to guidance or a little tooltip might be helpful. 

Another note that I haven't seen what this looks like, or what a11y looks and sounds like. Signing myself up for reviewing this, when I can get this to run locally

**Comment by jimleroyer** [comment]
> I think this is a valid option. Curious to release it and see if it clear things up on the support side. My main concern is that it might be unclear what the difference between sent and delivered is. Link to guidance or a little tooltip might be helpful.

+1 to proper guidance. Even now, it's not super clear. Having clear definitions of what the status means would be great. I know that @adriannelee was looking into this in a task on zenhub.

> Another note that I haven't seen what this looks like, or what a11y looks and sounds like. Signing myself up for reviewing this, when I can get this to run locally

This is an old PR now.. maybe I can migrate it to the newest changes, or I'd have to re-apply manually. Whatever works easiest, I can do. Unfortunately, without the preview apps, we can't demonstrate without merging to staging.

**Comment by amazingphilippe** [comment]
>This is an old PR now..
I merged main into this branch without too many conflicts yesterday. Is there anything else that needed to be done?

File: app/templates/partials/nav/gc_header_nav.html (hunk 1 above)
Comment (jimleroyer) [reply app/templates/partials/nav/gc_header_nav.html:93]: It's actually the permission for _See dashboard statistics_, in code speak. You can see how these translates over there:
https://github.com/cds-snc/notification-admin/blob/9f799673d41572314a094fa409f57f5a7163c659/app/models/roles_and_permissions.py#L22

**Comment by jimleroyer** [comment]
> I merged main into this branch without too many conflicts yesterday. Is there anything else that needed to be done?

Lots of testing, making sure the new status reflects everywhere these are reported, such as reports. That PR is a prototype just to see if that could be done easily.

**Comment by jimleroyer** [comment]
That was a nice prototype but let's close this and delete related branch.

**Pull request closed without merging**